HerPublicWebsite.UnitTests/BusinessLogic/Services/CsvFileCreatorTests.cs

```diff
@@ -116,20 +116,89 @@ public void CreateFileData_CalledWithMultipleReferralRequests_GeneratesExpectedF
     [TestCase("+441234567890", "441234567890")]
-    public void CreateFileData_CalledWithSpecialCharacters_GeneratesEscapedFileData(string nameInput, string expectedOutput)
+    public void CreateReferralRequestFileData_CalledWithSpecialCharacters_GeneratesEscapedFileData(string nameInput, string expectedOutput)
     {
         // Arrange
         var underTest = new CsvFileCreator();
         var referralRequest = new ReferralRequestBuilder(1).WithFullName(nameInput).Build();
         var referralRequests = new List<ReferralRequest>() { referralRequest };
 
         // Act
-        var data = underTest.CreateFileData(referralRequests);
+        var data = underTest.CreateReferralRequestFileData(referralRequests);
 
         // Assert
         var reader = new StreamReader(data, Encoding.UTF8);
         reader.ReadToEnd().Should().Be(
 "Referral date,Referral code,Name,Email,Telephone,Address1,Address2,Town,County,Postcode,UPRN,EPC Band,EPC confirmed by homeowner,EPC Lodgement Date,Is off gas grid,Household income band,Is eligible postcode,Tenure\r\n" +
 $"2023-01-01 13:00:01,DummyCode00001,{expectedOutput},contact1@example.com,00001 123456,Address 1 line 1,Address 1 line 2,Town1,County1,AL01 1RS,100 111 222 001,E,,2023-01-01 15:00:01,yes,Below £31k,no,Owner\r\n");
     }
+
+    [Test]
+    public void CreateReferralRequestOverviewFileData_CalledWithReferralRequest_GeneratesExpectedFileData()
+    {
+        // Arrange
+        var underTest = new CsvFileCreator();
+        var referralRequest = new ReferralRequestBuilder(1).WithCustodianCode("114").Build();
+        var referralRequests = new List<ReferralRequest>() { referralRequest };
+
+        // Act
+        var data = underTest.CreateReferralRequestOverviewFileData(referralRequests);
+        
+        // Assert
+        var reader = new StreamReader(data, Encoding.UTF8);
+        reader.ReadToEnd().Should().Be(
+"Consortium,Local Authority,Referral Code\r\n" +
+"Bristol,Bath and North East Somerset Council,DummyCode00001\r\n");
+    }
+    
+    [Test]
+    public void CreateReferralRequestOverviewFileData_CalledWithReferralRequestFromNullConsortium_GeneratesExpectedFileData()
+    {
+        // Arrange
+        var underTest = new CsvFileCreator();
+        var referralRequest = new ReferralRequestBuilder(1).WithCustodianCode("9052").WithFollowUp(new ReferralRequestFollowUpBuilder(1)).Build();
+        var referralRequests = new List<ReferralRequest>() { referralRequest };
+
+        // Act
+        var data = underTest.CreateReferralRequestOverviewFileData(referralRequests);
+        
+        // Assert
+        var reader = new StreamReader(data, Encoding.UTF8);
+        reader.ReadToEnd().Should().Be(
+            "Consortium,Local Authority,Referral Code\r\n" +
+            ",Aberdeenshire Council,DummyCode00001\r\n");
+    }
+    
+    [Test]
+    public void CreateReferralRequestFollowUpData_CalledWithReferralRequest_GeneratesExpectedFileData()
```

Comment:
I'm not a fan of how I've written this test. Maybe doing some c# equivalent of a list of the variables mapped to a referral request follow up builder would be better?

Similarly with the compared expected value, maybe it's worth doing the same as 612 and sticking it in a separate file?

Reply:
It's a bit repetitive, but I think it's still clear what you've done (which is more important)

You _could_ put the test data in separate files, but I'm not sure it adds too much.